spring-xd-spark-streaming-tests/src/test/java/org/springframework/xd/spark/streaming/AbstractSparkStreamingTests.java

```diff
@@ -108,12 +110,65 @@ public void testSparkProcessor() throws Exception {
 	}
 
 	@Test
+	public void testSparkProcessorWithInputType() throws Exception {
+		HttpSource source = new HttpSource(shell);
+		String streamName =  testName.getMethodName()  + new Random().nextInt();
+		FileSink sink = new FileSink().binary(true);
+		try {
+			String stream = String.format("%s | spark-word-count --inputType=text/plain | %s " +
+					"--inputType=text/plain", source, sink);
+			createStream(streamName, stream);
+			source.ensureReady().postData(TEST_MESSAGE);
+			assertThat(sink, XDMatchers.eventually(XDMatchers.hasContentsThat(equalTo("(foo,3)"))));
+		}
+		finally {
+			streamOps.destroyStream(streamName);
+			sink.cleanup();
+		}
+	}
+
+	@Test
+	public void testSparkProcessorWithOutputType() throws Exception {
+		HttpSource source = new HttpSource(shell);
+		String streamName =  testName.getMethodName()  + new Random().nextInt();
+		FileSink sink = new FileSink().binary(true);
+		try {
+			String stream = String.format("%s | spark-word-count --outputType=application/json | %s", source, sink);
+			createStream(streamName, stream);
+			source.ensureReady().postData(TEST_MESSAGE);
+			assertThat(sink, XDMatchers.eventually(XDMatchers.hasContentsThat(equalTo("{\"_1\":\"foo\",\"_2\":3}"))));
+		}
+		finally {
+			streamOps.destroyStream(streamName);
+			sink.cleanup();
+		}
+	}
+
```

Comment:
In general I prefer using the processing chain tests for testing a single processor which allow you to deploy a stream containing a processor under test which you can send and receive messages directly. This is much lighter weight and less brittle than deploying using an embedded shell, posting an http message and checking the contents of a file.  For testing a sink, using file output is appropriate, but for processor tests this is unnecessary.